### ISSUE TITLE
[New cop] Detect `<<` and `~` operators that is like heredoc

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#4573]
`<<~` heredoc has been usable since Ruby 2.3. However, the code is parsable by Ruby 2.2 or older. 
This behaviour is confusing.

See https://github.com/bbatsov/rubocop/issues/4572
https://bacon-cannon.herokuapp.com/parmlinks/b3eefefa-e5f5-4c0f-b6fa-6b80b32036d3

So, I think we should add a new cop for this problem.

--------

## Expected behavior

For the following code: 

```ruby
x <<~A
  foo
A
```

In Ruby 2.3 or higher: RuboCop does nothing.

In other Rubies: RuboCop says ``Add a whitespace to the right of the `~` if the operators are `<<` and `~`, or configure TargetRubyVersion by `.rubocop.yml` if it is a heredoc. ``(like Lint/AmbiguousOperator)

## Actual behavior

RuboCop does not work for the code.

## Steps to reproduce the problem

- Put the code
- Does not configure `TargetRubyVersion`

## RuboCop version

master

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
I'm wondering if we should have a check that `TargetRubyVersion` matches the project Ruby version from `.ruby-version` or `Gemfile`. It seems like a higher level concern than with just this cop. WDYT?

**Comment by rrosenblum** [comment]
@drenmi The purpose of the `TargetRubyVersion` is to be an override to the configured or running version of Ruby. In RuboCop's own case, the `TargetRubyVersion` is set to 2.0 because that is the lowest Ruby version that we support. I am developing RuboCop using Ruby 2.4. Checking that the `TargetRubyVersion` matches the `.ruby-version` or `Gemfile` is likely a mistake.

Please also see the conversation in #4186.

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
